python/ray/data/_internal/execution/operators/actor_pool_submitter.py

```diff
@@ -0,0 +1,186 @@
+from typing import Dict, Any, Iterator, Callable, List, Tuple
+import ray
+from ray.data.block import Block, BlockAccessor, BlockMetadata, BlockExecStats
+from ray.data.context import DEFAULT_SCHEDULING_STRATEGY, DatasetContext
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.execution.operators.map_task_submitter import MapTaskSubmitter
+from ray.types import ObjectRef
+
+
+class ActorPoolSubmitter(MapTaskSubmitter):
+    """A task submitter for MapOperator that uses a Ray actor pool."""
+
+    def __init__(
+        self,
+        transform_fn_ref: ObjectRef[Callable[[Iterator[Block]], Iterator[Block]]],
+        ray_remote_args: Dict[str, Any],
+        pool_size: int,
```

Comment:
Based on the role as a "submitter", it looks better to take in an already created ActorPool here. The resource manager is the ActorPool not the submitter.

Reply:
The complicated bit here is that we need to wait until the operator has actually started executing before we create the actors for the pool, which is currently determined in `MapOperatorState.add_input()`. We can pass in an empty actor pool and still have `ActorPoolSubmitter.start()` add all of the actors, but that would still require the `pool_size` and `ray_remote_args` arguments.

I'm thinking about submitting a follow-up PR that consolidates `MapOperatorState` into `MapOperator` (since the latter is essentially a passthrough interface to the former so there's not much use in have separate abstractions) and proposes the addition of a `PhysicalOperator.start()` API, at which point pulling out the `ActorPool` can be made a bit cleaner; could we defer this until that refactor?